### DISCUSSION TITLE
plotting the objective function

### DISCUSSION THREAD

**Discussion by nilkiweera** [yalmip/YALMIP#938] [Optimization modelling] [answered]
Hi Johan,

Hope you are well, 

I have a small question regarding plotting the objective function. 

I have a linear objective function comprised of four decision variables, the objective is to minimize the error between a setpoint  (Q_max_power) and manipulated variables (u{k})
objective = objective + norm(R1*(u{k}(2) - Q_max_power{k}(1)),1) ...

      + norm(R4*(u{k}(8) - Q_max_power{k}(4)),1) + norm(R2*(u{k}(4) - Q_max_power{k}(2)),1) ...

      + norm(R3*(u{k}(6) - Q_max_power{k}(3)),1);

I want to plot the objective function with respect to all the variables and see how it evolves, this is solved through gurobi. Thank you so much,
Nilki

**Comment by johanlofberg** [comment, marked as answer]
all variables? more than two will make it tricky...

general plotting of expressions is not possible, but if you mean you have an optimizer object and you want to plot the optimal objective w.r.t 2d-parameters, there are example here https://yalmip.github.io/optimizerupdates

**Reply by nilkiweera** [reply]
Hi Johan,
thank you very much i will look into that  :)
regards,
Nilki
